src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,156 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2020 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+from scipy.linalg import eig
+from pymor.algorithms.samdp import samdp
+from pymor.core.base import BasicObject
+from pymor.models.iosys import LTIModel
+from pymor.vectorarrays.numpy import NumpyVectorArray
+from pymor.parameters.base import Mu
+from pymor.reductors.basic import LTIPGReductor
+from pymor.algorithms.gram_schmidt import gram_schmidt, gram_schmidt_biorth
+import numpy as np
+
+
+class MTReductor(BasicObject):
+    """Modal Truncation reductor.
+
+    See Section 9.2 in [A05]_.
+
+    Parameters
+    ----------
+    fom
+        The full-order |LTIModel| to reduce.
+    mu
+        |Parameter values|.
+    """
+
+    def __init__(self, fom, mu=None):
+        assert isinstance(fom, LTIModel)
+        if not isinstance(mu, Mu):
+            mu = fom.parameters.parse(mu)
+        assert fom.parameters.assert_compatible(mu)
+        self.fom = fom
+        self.mu = mu
+        self.V = None
+        self.W = None
+        self._pg_reductor = None
+
+    def reduce(self, r=None, decomposition='samdp', projection='orth',
+               symmetric=False, method_options=None):
+        """Modal Truncation.
+
+        Parameters
+        ----------
+        r
+            Order of the reduced model
+        decomposition
+            Algortihm to use for the decomposition:
+
+            -`'eigs'`: use scipy.linalg.eig algorithm
+            -`'samdp'`: find dominant poles using samdp algorithm
+        projection
+            Projection method:
+            - `'orth'`: projection matrices are orthogonalized with
+              respect to the Euclidean inner product
+            - `'biorth'`: projection matrices are biorthogolized with
+              respect to the E product
+        symmetric
+            True if Operator A is symmetric, False if not
+        method_options
+            optional dict with more options for the method used to calculate
+            eigenvalues and eigenvectors
+
+        Returns
+        -------
+        rom
+            Reduced-order model.
+        """
+        assert min(self.fom.B.source.dim, self.fom.C.range.dim) <= r < self.fom.order
+        assert projection in ('orth', 'biorth')
+        assert decomposition in ('eigs', 'samdp')
+        assert method_options is None or isinstance(method_options, dict)
+        if not method_options:
+            method_options = {}
+            method_options['which'] = 'LR'
+
+        self.V = self.fom.B.as_range_array().empty(reserve=r)
+        self.W = self.fom.C.as_source_array().empty(reserve=r)
+
+        if decomposition == 'eigs':
+            if self.fom.A.sparse:
+                A = self.fom.A.as_source_array().to_numpy()
+            else:
+                A = self.fom.A.matrix
```

Comment:
It is more convenient to use `to_matrix` from `pymor.algorithms.to_matrix` to convert an `Operator` to a matrix.

Also, large `Operators` should not be converted silently to dense matrices, or at all. See how [`LTIModel.poles`](https://github.com/pymor/pymor/blob/2f7955ac6df214fbe1cb5d7b130c5d00e6d9bd9a/src/pymor/models/iosys.py#L621-L626) does it.